cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+};
+
+const setSearchAndSaveAndVerify = (config) => {
+  navigateToDiscoverPage();
+  setDataset(config.dataset);
+  cy.setQueryLanguage(config.language);
+  setDatePickerDatesAndSearchIfRelevant(config.language);
+
+  setSearchConfigurations({
+    addFilter: config.filters,
+    queryString: prepareQueryStringForEditor(config.queryString, config.language, config.dataset),
+    setHistogramInterval: config.histogram,
+    selectFields: config.selectFields,
+    applySort: config.sort,
+  });
+  verifyDiscoverPageState({
+    dataset: config.dataset,
+    queryString: config.queryString,
+    language: config.language,
+    hitCount: config.hitCount,
+    checkFilter: config.filters,
+    checkHistogramInterval: config.histogram,
+    checkSelectedField: config.selectFields,
+    verifyTableData: config.sampleTableData,
+  });
+  saveSearch({ searchName: config.saveName });
+
+  // There is a small chance where if we go to assets page,
+  // the saved search does not appear. So adding this wait
+  cy.wait(1000);
+
+  verifySavedSearchInAssetsPage({
+    dataset: config.dataset,
+    searchName: config.saveName,
+    queryString: config.queryString,
+    language: config.language,
+    checkHistogramInterval: config.histogram,
+    checkSelectedField: config.selectFields,
+    checkSort: config.sort,
+    checkFilter: config.filters,
+  });
+};
+
+const loadSavedSearchAndVerify = (config) => {
+  // We are starting from various languages
+  // to guard against: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/9078
+  ['DQL', 'Lucene', 'OpenSearch SQL', 'PPL'].forEach((startingLanguage) => {
+    // TODO: Remove this line once bugs are fixed
+    // https://github.com/opensearch-project/OpenSearch-Dashboards/issues/9078
+    if (startingLanguage !== config.language) return;
+
+    navigateToDiscoverPage();
+    cy.getElementByTestId('discoverNewButton').click();
+
+    // Intentionally setting INDEX_PATTERN dataset here so that
+    // we have access to all four languages that INDEX_PATTERN allows.
+    // This means that we are only testing loading a saved search
+    // starting from an INDEX_PATTERN dataset, but I think testing where the
+    // start is a permutation of other dataset is overkill
+    setDataset(INDEX_PATTERN_WITH_TIME);
+
+    cy.setQueryLanguage(startingLanguage);
+    loadSavedSearch(config.saveName);
+    setDatePickerDatesAndSearchIfRelevant(config.language);
+    verifyDiscoverPageState({
+      dataset: config.dataset,
+      queryString: config.queryString,
+      language: config.language,
+      hitCount: config.hitCount,
+      checkFilter: config.filters,
+      checkHistogramInterval: config.histogram,
+      checkSelectedField: config.selectFields,
+      verifyTableData: config.sampleTableData,
+    });
+  });
+};
+
+export const runSavedSearchCreateTests = () => {
+  describe('saved search', () => {
+    beforeEach(() => {
+      // Load test data
+      cy.setupTestData(
```

Comment:
Do we really need to reIndex the files, addDatasources , createWorkspace for each of the individual test cases. Since all of these constructs are saved in OpenSearch which are not refreshed across the tests even with the test isolation being turned on. 

We can isolate the UI navigation/interaction, opening up pages and add it in the beforeEach. While the above operations can be doen only once per test suites and removed at the end of the entire test suite.

Reply:
we synced offline, but to reiterate here, we have concluded that we need to use "beforeEach" and "afterEach"